codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -77,6 +84,17 @@ public DefaultHttp2ConnectionDecoder(Http2Connection connection,
         connection.local().flowController().frameWriter(encoder.frameWriter());
     }
 
+    public DefaultHttp2ConnectionDecoder(Http2Connection connection,
+                                         Http2ConnectionEncoder encoder,
+                                         Http2FrameReader frameReader,
+                                         Http2Settings initialSettings,
+                                         Http2PromisedRequestVerifier requestVerifier) throws Http2Exception {
+        this(connection, encoder, frameReader, requestVerifier);
+        if (initialSettings != null) {
+            applyLocalSettings(initialSettings);
```

Comment:
Not sure I follow why we're doing this ... Maybe I'm not getting the point of this PR. It seems that we can't actually apply the settings until we receive an ack, right?  Specifically, any threshold that the remote endpoint might violate before receiving the settings frame (i.e. maxConcurrentStreams, frameSize, etc.).